tests/unit/multi-currency/test-class-storefront-integration.php

```diff
@@ -0,0 +1,106 @@
+use WCPay\MultiCurrency\StorefrontIntegration;
+
+/**
+ * WCPay\MultiCurrency\StorefrontIntegration unit tests.
+ */
+class WCPay_Multi_Currency_Storefront_Integration_Tests extends WP_UnitTestCase {
+	/**
+	 * Mock MultiCurrency.
+	 *
+	 * @var MultiCurrency|PHPUnit_Framework_MockObject_MockObject
+	 */
+	private $mock_multi_currency;
+
+	/**
+	 * Pre-test setup
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->mock_multi_currency    = $this->createMock( MultiCurrency::class );
+		$this->storefront_integration = new StorefrontIntegration( $this->mock_multi_currency );
+	}
+
+	public function tearDown() {
+		remove_all_filters( 'option_wcpay_multi_currency_enable_storefront_switcher' );
+
+		parent::tearDown();
+	}
+
+	public function test_register_settings_on_init() {
+		$this->assertSame(
+			10,
+			has_filter( 'wcpay_multi_currency_enabled_currencies_settings', [ $this->storefront_integration, 'filter_store_settings' ] ),
+			"The filter 'wcpay_multi_currency_enabled_currencies_settings' with function 'filter_store_settings' was not registered with a priority of 10."
+		);
+	}
+
+	/**
+	 * @dataProvider switcher_filter_provider
+	 */
+	public function test_does_not_register_actions_when_switcher_disabled( $filter, $function_name ) {
+		$this->mock_option( 'no' );
+		// Reinit class to re-evaluate conditional hooks.
+		$this->storefront_integration = new StorefrontIntegration( $this->mock_multi_currency );
+
+		$this->assertFalse(
+			has_filter( $filter, [ $this->storefront_integration, $function_name ] ),
+			"The filter '$filter' with function '$function_name' was found."
+		);
+	}
+
+
+	/**
+	 * @dataProvider switcher_filter_provider
+	 */
+	public function test_registers_actions_when_switcher_default( $filter, $function_name ) {
+		$this->mock_option( '' );
+		// Reinit class to re-evaluate conditional hooks.
+		$this->storefront_integration = new StorefrontIntegration( $this->mock_multi_currency );
+
+		$this->assertGreaterThan(
+			10,
+			has_filter( $filter, [ $this->storefront_integration, $function_name ] ),
+			"The filter '$filter' with function '$function_name' was not registered with a priority above 10."
+		);
+	}
+
+	/**
+	 * @dataProvider switcher_filter_provider
+	 */
+	public function test_registers_actions_when_switcher_enabled( $filter, $function_name ) {
+		$this->mock_option( 'yes' );
+		// Reinit class to re-evaluate conditional hooks.
+		$this->storefront_integration = new StorefrontIntegration( $this->mock_multi_currency );
+
+		$this->assertGreaterThan(
+			10,
+			has_filter( $filter, [ $this->storefront_integration, $function_name ] ),
+			"The filter '$filter' with function '$function_name' was not registered with a priority above 10."
+		);
+	}
+
+	public function switcher_filter_provider() {
+		return [
+			[ 'woocommerce_breadcrumb_defaults', 'modify_breadcrumb_defaults' ],
+			[ 'wp_enqueue_scripts', 'add_inline_css' ],
+		];
+	}
+
+	private function mock_option( $value ) {
```

Comment:
Would you mind adding a more descriptive name here? Or maybe pass the option name as an argument for this method? Without this context, it's a bit hard to know which option is being mocked by scanning the tests' code.

Reply:
Removed this function in https://github.com/Automattic/woocommerce-payments/pull/2471/commits/dfa75a48698a9f2119214a863447450c49da5139 due to when fixing the get_option call in your previous comment, it broke the tests.